File: src/main/java/io/tripled/social/client/presentation/controller/FollowController.java
package io.tripled.social.client.presentation.controller;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

import io.tripled.social.client.application.FollowUserUseCase;
import io.tripled.social.client.presentation.Output;

public class FollowController extends AbstractController {

  private FollowUserUseCase followUserUseCase;

  public FollowController(FollowUserUseCase followUserUseCase) {
    this.followUserUseCase = followUserUseCase;
  }

  @Override
  protected Pattern getPattern() {
    return Pattern.compile("(.+) follows (.+)");
  }

  @Override
  public void execute(String inputLine, Output output) {
    Matcher matcher = getPattern()
        .matcher(inputLine.trim());

    if (matcher.matches()) {
      String userName = matcher.group(1).trim();
      String userNameToFollow = matcher.group(2).trim();

      output.print(followUserUseCase.follow(userName, userNameToFollow));
    }
  }
}

File: src/main/kotlin/io/tripled/social/client/presentation/controller/FollowController.kt
package io.tripled.social.client.presentation.controller

import io.tripled.social.client.application.FollowUserUseCase
import io.tripled.social.client.presentation.Controller
import io.tripled.social.client.presentation.Output
import io.tripled.social.client.presentation.request.FollowCliRequest

class FollowController(private val followUserUseCase: FollowUserUseCase) : Controller<FollowCliRequest> {
    override fun execute(request: FollowCliRequest, output: Output) {
        output.print(followUserUseCase.follow(request.userName, request.userNameToFollow))
    }
}
